geode-core/src/main/java/org/apache/geode/internal/cache/control/HeapMemoryMonitor.java

```diff
@@ -738,14 +738,19 @@ public void run() {
   }
 
   protected Set<DistributedMember> getHeapCriticalMembersFrom(Set<DistributedMember> members) {
-    Set<DistributedMember> result = new HashSet(members);
-    if (!this.mostRecentEvent.getState().isCritical()) {
-      result.remove(cache.getMyId());
-    }
-    result.retainAll(resourceAdvisor.adviseCriticalMembers());
+    Set<DistributedMember> result = new HashSet<>(members);
```

Comment:
These three lines can be changed to:
  return getCriticalMembers().retainAll(members);
and creates one HashSet instead of two.